src/test/java/org/apache/commons/collections4/iterators/ArrayListIteratorTest.java

```diff
@@ -104,15 +105,8 @@ public void testListIteratorSet() {
         // a call to set() before a call to next() or previous() should throw an IllegalStateException
         iter = makeArrayListIterator(testArray);
 
-        try {
-            iter.set((E) "should fail");
-            fail("ListIterator#set should fail if next() or previous() have not yet been called.");
-        } catch (final IllegalStateException e) {
-            // expected
-        } catch (final Throwable t) { // should never happen
-            fail(t.toString());
-        }
-
+        ListIterator<E> finalIter = iter;
+        assertThrows(IllegalStateException.class, () -> finalIter.set((E) "should fail"));
```

Comment:
Would be good to preserve this error message too, IMO.

Reply:
ditto